Sources/XcodeGenKit/PBXProjGenerator.swift

```diff
@@ -436,40 +436,60 @@ public class PBXProjGenerator {
         }
     }
 
-    func getGroups(path: Path, depth: Int = 0) throws -> (filePaths: [Path], groups: [PBXGroup]) {
+    func getGroups(path: Path, depth: Int = 0) throws -> (sourceFiles: [SourceFile], groups: [PBXGroup]) {
 
         let excludedFiles: [String] = [".DS_Store"]
         let directories = try path.children().filter { $0.isDirectory && $0.extension == nil && $0.extension != "lproj" }
         let filePaths = try path.children().filter { $0.isFile || $0.extension != nil && $0.extension != "lproj" }.filter { !excludedFiles.contains($0.lastComponent) }
         let localisedDirectories = try path.children().filter { $0.extension == "lproj" }
-        var groupChildren: [String] = []
-        var allFilePaths: [Path] = filePaths
+        var groupChildren: [String] = filePaths.map { getFileReference(path: $0, inPath: path) }
+        var allSourceFiles: [SourceFile] = filePaths.map { generateSourceFile(path: $0) }
         var groups: [PBXGroup] = []
 
         for path in directories {
             let subGroups = try getGroups(path: path, depth: depth + 1)
-            allFilePaths += subGroups.filePaths
+            allSourceFiles += subGroups.sourceFiles
             groupChildren.append(subGroups.groups.first!.reference)
             groups += subGroups.groups
         }
 
-        for filePath in filePaths {
-            let fileReference = getFileReference(path: filePath, inPath: path)
-            groupChildren.append(fileReference)
+        // create variant groups of the base localisation first
+        var baseLocalisationVariantGroups = [PBXVariantGroup]()
```

Comment:
minor: the rest of the codebase uses `var baseLocalisationVariantGroups: [String] = []` syntax